tests/e2e/gc_rwx_statefulsets.go

```diff
@@ -0,0 +1,561 @@
+		for _, sspod := range ssPodsAfterScaleUp.Items {
+			err := fpod.WaitForPodsReady(client, statefulset.Namespace, sspod.Name, 0)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pod, err := client.CoreV1().Pods(namespace).Get(ctx, sspod.Name, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, volumespec := range pod.Spec.Volumes {
+				if volumespec.PersistentVolumeClaim != nil {
+					persistentvolume := getPvFromClaim(client, statefulset.Namespace,
+						volumespec.PersistentVolumeClaim.ClaimName)
+
+					volHandle := persistentvolume.Spec.CSI.VolumeHandle
+					gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+					volumeID := getVolumeIDFromSupervisorCluster(volHandle)
+					gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+
+					ginkgo.By("Verify scale up operation should not introduced new volume")
+					gomega.Expect(contains(volumesBeforeScaleDown, volHandle)).To(gomega.BeTrue())
+
+					ginkgo.By("Verifying whether the CnsFileAccessConfig CRD is created or not for Pod")
+					verifyCNSFileAccessConfigCRDInSupervisor(ctx, f, sspod.Spec.NodeName+"-"+volHandle,
+						crdCNSFileAccessConfig, crdVersion, crdGroup, true)
+
+					cnsFileAccessConfigCRDList = append(cnsFileAccessConfigCRDList, sspod.Spec.NodeName+"-"+volHandle)
+				}
+			}
+		}
+
+		replicas = 0
+		ginkgo.By(fmt.Sprintf("Scaling down statefulsets to number of Replica: %v", replicas))
+		_, scaledownErr = fss.Scale(client, statefulset, replicas)
+		gomega.Expect(scaledownErr).NotTo(gomega.HaveOccurred())
+		fss.WaitForStatusReplicas(client, statefulset, replicas)
+		ssPodsAfterScaleDown = fss.GetPodList(client, statefulset)
+		gomega.Expect(len(ssPodsAfterScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset should match with number of replicas")
+
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
+		isSTSDeleted = true
+
+		ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+		err = client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isServiceDeleted = true
+
+		ginkgo.By("Wait and verify PVC is fully deleted")
+		for _, volume := range volumesBeforeScaleDown {
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volume)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Wait and verify CNSFileAccessConfig CRD is fully deleted")
+		for _, crdName := range cnsFileAccessConfigCRDList {
+			verifyCNSFileAccessConfigCRDInSupervisor(ctx, f, crdName, crdCNSFileAccessConfig,
+				crdVersion, crdGroup, false)
+		}
+	})
+
+	/*
+		Test to verify file volume provision with statefulsets
+		1. Create Storage class
+		2. Create Nginx service
+		3. Create Nginx statefulset with 3 replicas and
+				podManagementPolicy=Parallel  using the Storage
```

Comment:
so we seem to run the same setup and same set of validations twice barring this one step.

Why dont we just create 2 statefulsets one w/ OrderedReady and another w/ Parallel and make all verifications clubbed, if it fails for either of them we can fail the test? Would it not save time?

Reply:
Separate tests are easy to do code maintenance and moreover, most of the clean-up operations in golang are present in-between the codes and get tricky to handle multiple conditions.

Keeping it is as is to map to TDS.